pyfolio/tears.py

```diff
@@ -1536,9 +1536,15 @@ def create_perf_attrib_tear_sheet(returns,
     perf_attrib.show_perf_attrib_stats(returns, positions, factor_returns,
                                        factor_loadings, transactions)
 
-    # one section for the returns plot, and for each factor grouping
-    # one section for factor returns, and one for risk exposures
-    vertical_sections = 1 + 2 * len(factor_partitions)
+    # only plot partitions for which there are data in perf_attrib_data
+    num_partitions = 0
+    for partitions in factor_partitions.values():
+        if len(perf_attrib_data.columns.intersection(partitions)) > 0:
+            num_partitions += 1
+
+    # one section for the returns plot, and for each factor partition,
+    # one section for factor returns, and one for exposures
+    vertical_sections = 1 + 2 * max(num_partitions, 1)
```

Comment:
Why is this max(num_partitions, 1)?  If we have no valid partitions, what happens in those two entries?

Reply:
then we just plot all attributed returns on one plot, and all factor exposures on one plot